docker/docker.ts

```diff
@@ -307,16 +344,18 @@ async function buildImageAsync(
            `No digest available for image ${imageName}`, logResource);
     }
 
-    return {
-        id: inspectResult.trim(),
-        stages: stages,
-    };
+    // Trim off the hash kind if there is one.
+    let imageId = inspectResult.trim();
+    const colonIndex = imageId.lastIndexOf(":");
+    imageId = colonIndex < 0 ? imageId : imageId.substr(colonIndex + 1);
```

Comment:
Why do we trim this off?  Shouldn't we return the full `imageID` just so that we expose exactly what Docker uses as the image ID?

Reply:
ImageID is "algorithm:hex".  I trim off the `algorithm:` part since we can't have colons in the tag portion.  As we want just a unique tag, the hex portion is sufficient.  I could include the algorithm.  but it seems really superfluous... 

Thoughts?